webui/react/src/hooks/useSettings.ts

```diff
@@ -66,12 +66,12 @@ const queryParamToType = <T>(
 ): Primitive | undefined => {
   if (param === null || param === undefined) return undefined;
   if (type.is(false)) return param === 'true';
-  if (type.is(0)) {
+  if (type.is(0) || type.is([0])) {
```

Comment:
this code is a bit confusing to me. prob its nice to refactor sooner or later.
whats the reason why we use the constant values like `ype.is(0) || type.is([0]` instead of `type === boolean` etc?